### DISCUSSION TITLE
Creating awesome-fprime repository in F´ Community Organization

### DISCUSSION THREAD

**Discussion by ThibFrgsGmz** [nasa/fprime#977] [General]
Hi there 👋,

How about creating an "awesome-fprime" repository in the [F´ Community Organization](https://github.com/fprime-community) ?

This repository would have the same function as the traditional "awesome-xyz" repository which gathers the best resources for a specific programming language, framework, platform, etc.

In other words, we would find in this repository an exhaustive list of all Open Source projects using F Prime. This would make it easier to find projects based on F Prime to study how others implement F Prime.

To get an idea, here are some examples of awesome list implementations:
- awesome-vue.js: https://github.com/vuejs/awesome-vue
- awesome-python: https://github.com/vinta/awesome-python
- awesome-iOS: https://github.com/vsouza/awesome-ios.

**Comment by astroesteban** [comment]
This is a great idea! The first thing I often do is search GitHub for awesome-xyz and just scroll through to get a list of blogs, tutorials, newsletters, videos, and other related projects. This would be great to have for F' as well!

**Comment by LeStarch** [comment]
This is a cool Idea.  Typically who does the curation?  The project itself?  The user community?

**Reply by ThibFrgsGmz** [reply]
Traditionally it is the community that adds projects to the list through the PR mechanism. Even if the project maintainers have established a small list at the beginning.

Then the PR is accepted if the proposed project sufficiently meets the established acceptance criteria.

In my opinion there should be acceptance criteria so that there are no "test" projects in the list that have no added value compared to the tutorial projects.
The goal is really to help and motivate the community by listing projects that implement F´ in an interesting way, on a particular hardware platform, or for various missions/applications.

Also there should be a PR template to respect when requesting an addition so as not to add too much burden to the repo maintainers.

I think the awesome-vue PR template I put in my main post is a good place to start.
If I quickly adapt it to F´, it would look like this (not very convinced of all the fields but it's a first try):

---

### `General`
> ✏️ Mark the necessary items without changing the structure of the PR template.

- [ ] Pull request template structure not broken

### `Type`

> ℹ️  What types of feature does the project introduce?

> 👉 _Put an `x` in the boxes that apply_

- [ ] Embedded HW platform
- [ ] New-space application 
- [ ] University project
- [ ] Professional project

### `Checklist`

> ℹ️  Check all checkboxes - this will indicate that you have done everything in accordance with the rules in [CONTRIBUTING].

> 👉  _Put an `x` in the boxes that apply._

- [ ] Title as described
- [ ] Make sure you put things in the right category
- [ ] Always add your items to the end of a list

#### `Open Source`

- [ ] Link description does not contain a link to an author / third-party resource
- [ ] The documentation (README) contains a description of the project, illustration of the project with a demo or screenshots and a CONTRIBUTING section
- [ ] The documentation is in English.
- [ ] The project is active and maintained.
- [ ] The project accepts contributions.
- [ ] Not a commercial product

#### `Project`

- [ ] The project is using F´ intensively as submodule or not.
- [ ] The project is original and not too simple. For that reason, test and tutorial projects may rejected.



---

**Comment by LeStarch** [comment]
Any chance we could start collecting projects here?  When we make the REPO, we'd like to have some projects initially such that it isn't empty.

**Reply by SterlingPeet** [reply]
Considering the docs already point at my component cookiecutter repository as an example of an external cookiecutter extension for ``fprime-util new``, you could start with that project.

**Reply by ThibFrgsGmz** [reply]
If I search for the word "fprime" on GitHub and quickly list the projects that look promising I would say:

* [fprime-doorbell](https://github.com/pelmini/fprime-doorbell) by @pelmini but which should incorporate a README.md file to describe its project.
* [FS](https://github.com/FP-Mission/FS) by @lewisjaggi who could provide a short description to put in the repo.
* [Fprime-CalPoly](https://github.com/Reconfigurable-Computing-CalPoly-Pomona/Fprime-CalPoly) by @kylewong94 but there is no source code. See if it's relevant to reference it. 
* [fprime-GT1-Payload](https://github.com/SterlingPeet/fprime-GT1-Payload) by @SterlingPeet
* [r-prime](https://github.com/BEARUBC/r-prime) by @raunakab which seems to be a project under development of an implementation of F Prime in Rust, but not sure if it has its place in the repo except in a very specific section.
* [gps-application](https://github.com/LeStarch/gps-application) by @LeStarch I don't know if your project has any added value to the gps tutorial.
* [quest-fw](https://github.com/quest-fw/quest-fw) by @genemerewether, reading the README diagonally I don't know the mission of the project so the presentation could be improved to reference it.
* [Fprime_MSTAR](https://github.com/Reconfigurable-Computing-CalPoly-Pomona/Fprime_MSTAR) whose first contributor is @mcnerd16. There is no source code so I don't know if it is interesting to reference it or to put it in a special section.

Maybe we should reference the projects of the fprime-community organization like the [fprime-arduino](https://github.com/fprime-community/fprime-arduino).

We can also reference the [fprime/RPI](https://github.com/nasa/fprime/tree/devel/RPI) deployment or take it out of the framework, put it in fprime-community and reference it in the "awesome" repo.

_Note:_ Btw, now that I think about it, why are there two deployments in the default framework (Ref and RPI)? 🤔

**Reply by LeStarch** [reply]
This is a good list as a basis.  Thanks for digging those up.  Anyone esle have suggestions?

To answer your specific questions:

1. gps-application does not add anything.  I may be working on a baremetal implementation there, but the value to the community was the GPS tutorial that you transferred.   I do have another repo, fprime-rc-car, or something like that, that will add value....but it is currently under a lot of refactor....so not yet.
2. Ref is a general reference app design to show off F´ and its components.  RPI is specifically tailored to a low cost embedded system to show off the embedded context of fprime.

**Reply by SMorettini** [reply]
There is any news about this discussion? I would also find interesting and helpful to have the repo.

**Reply by thomas-bc** [reply]
@SMorettini the repo is there: https://github.com/fprime-community/fprime-awesome, feel free to make PRs to it (and tag us for review)

Although it hasn't been updated in a while, we may want to promote it a little bit in some way

**Comment by thomas-bc** [comment]
Since this was brought back up, I'll start another thread.

There is somewhat of an overlap of objectives in the following:
- [Projects](https://nasa.github.io/fprime/projects.html) is a list of projects that are using F´. It is currently only listing JPL missions, but I don't think we intend to restrict it to that
- [F´ Community README](https://github.com/fprime-community) highlights useful repositories in F´ Community
- https://github.com/fprime-community/fprime-awesome lists all sorts of things but seems to have gone stale.

We may want to consolidate our guideline here, or reference https://github.com/fprime-community/fprime-awesome in more places if we intend for it to be a source too. 

Thoughts?

**Reply by codeflight1** [reply]
I haven't updated `fprime-awesome` since the repository creation, but I'm happy to continue expanding it if there is a list of projects to add.

**Reply by thomas-bc** [reply]
Maybe a good idea in `fprime-awesome` would be to reference the [F´ Community README](https://github.com/fprime-community) for all of the F´-supported things (references, tutorials etc...) and then let the community add their own projects in `fprime-awesome`? That way we don't duplicate the information, and the F´ Community README won't go stale.

**Reply by thomas-bc** [reply]
Could also reference https://nasa.github.io/fprime/projects.html as a "What's flying" section haha

I encourage external projects to make PR to https://github.com/nasa/fprime/blob/devel/docs/projects.md if they want to be added there! Could also add a POC column if wanted.

**Reply by codeflight1** [reply]
`fprime-awesome` currently references the projects page as the Missions section.

I like the idea of keeping `fprime-awesome` just for community projects, I was unaware of the community readme when I created the `fprime-awesome` list.
